xarray/backends/h5netcdf_.py

```diff
@@ -183,10 +183,11 @@ def prepare_variable(self, name, variable, check_encoding=False,
 
         return target, variable.data
 
-    def sync(self):
+    def sync(self, compute=True):
         with self.ensure_open(autoclose=True):
-            super(H5NetCDFStore, self).sync()
+            super(H5NetCDFStore, self).sync(compute=compute)
             self.ds.sync()
+        return
```

Comment:
This sync logic feels repetitive across the different backend classes. Maybe there is a way to refactor a bit to avoid redundancy.

Reply:
I totally agree. I'd like to get this to work for h5netcdf before I embark on a refactor though.

Reply:
an empty `return` is a pointless if it's the last nice of a function -- you should remove these.